File: hw/nase/u3/hw21AnimalFarm/JavaScript/model/farmAnimal.js
const { Food } = require('./infrastructure');

class FarmAnimal {
    constructor(id, breed, bornOn, weight) {
        if (this.constructor === FarmAnimal) {
        }
        this._id = id;
        this._breed = breed;
        this._bornOn = new Date(bornOn);
        this._weight = weight;

        this.cuts = [];
        this.product = null;
        this.slaughterHouse = null;
    }

    getAgeInMonths() {
        const today = new Date();
        const yearsDiff = today.getFullYear() - this._bornOn.getFullYear();
        const monthsDiff = today.getMonth() - this._bornOn.getMonth();
        return (yearsDiff * 12) + monthsDiff;
    }

    feed(food) {
        return `The purebred animal ${this._breed} has been fed with: ${food._description}`;
    }
}

module.exports = FarmAnimal;

File: hw/nase/u3/hw21AnimalFarm/Python/model/farm_animal.py
from abc import ABC
from datetime import date
from typing import List, Optional
from model.infrastructure import Cut, Food, Product

class FarmAnimal(ABC):
    def __init__(self, id_animal: int, breed: str, born_on: date, weight: float):
        self._id = id_animal
        self._breed = breed
        self._born_on = born_on
        self._weight = weight
        
        self.cuts: List[Cut] = []
        self.product: Optional[Product] = None
        self.slaughter_house: Optional[str] = None

    def get_age_in_months(self) -> int:
        today = date.today()
        return (today.year - self._born_on.year) * 12 + today.month - self._born_on.month

    def feed(self, food: Food) -> str:
        return f"The purebred animal {self._breed} has been fed with: {food._description}"